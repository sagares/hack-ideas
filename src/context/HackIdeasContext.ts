import { createContext } from "react";
import { IdeasArray } from "../HackIdea.types";

type Employee = {
  empId: string;
};

type HackIdeasContextType = {
  currentUser: Employee;
  logout: () => void;
  setCurrentUser: (employee: Employee) => void;
  ideas: IdeasArray;
  setIdeas: (ideas: IdeasArray) => void;
};

const HackIdeasContext = createContext<Partial<HackIdeasContextType>>({});

export { HackIdeasContext };
