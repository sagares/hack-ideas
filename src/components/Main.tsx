import React, { useContext } from "react";
import HackIdeasContext from "../context";
import { HomePage, LoginForm } from "./";

const Main = () => {
  const { currentUser } = useContext(HackIdeasContext);

  return !currentUser ? <LoginForm /> : <HomePage />;
};

export default Main;
