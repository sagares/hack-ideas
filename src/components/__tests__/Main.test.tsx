/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import HackIdeasContext from "../../context";
import Main from "../Main";

describe("Main Component", () => {
  let element, component, setCurrentUser;

  beforeEach(() => {
    setCurrentUser = jest.fn().mockImplementation(() => {});
    element = document.createElement("div");
  });

  afterEach(() => {
    document.body.removeChild(element);
  });

  test("renders login form if current user is not defined", () => {
    component = render(
      <HackIdeasContext.Provider value={{ currentUser: null, setCurrentUser }}>
        <Main />
      </HackIdeasContext.Provider>,
      {
        container: document.body.appendChild(element),
      }
    );

    expect(component.container).toMatchSnapshot();
    expect(element.querySelector(".login-form-card")).toBeDefined();
  });

  test("renders home page if current user is defined", () => {
    component = render(
      <HackIdeasContext.Provider
        value={{ currentUser: { empId: "123456" }, setCurrentUser }}
      >
        <Main />
      </HackIdeasContext.Provider>,
      {
        container: document.body.appendChild(element),
      }
    );

    expect(component.container).toMatchSnapshot();
    expect(element.querySelector(".hack-ideas--home-page")).toBeDefined();
  });
});
