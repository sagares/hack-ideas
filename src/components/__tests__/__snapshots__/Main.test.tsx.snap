// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders home page if current user is defined 1`] = `
<div>
  <div
    class="hack-ideas--home-page"
  >
    <nav>
      <div
        class="nav-wrapper"
      >
        <a
          class="brand-logo"
        >
          { Hack Ideas }
        </a>
        <ul
          class="right hide-on-med-and-down"
        >
          <li>
            <a
              class="logout-btn"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="hack-ideas--filter row"
    >
      <div
        class="search-bar col s12 m12 l4"
      >
        <nav>
          <div
            class="nav-wrapper"
          >
            <form>
              <div
                class="input-field grey lighten-4"
              >
                <input
                  id="search"
                  type="search"
                  value=""
                />
                <label
                  class="label-icon"
                  for="search"
                >
                  <i
                    class="material-icons"
                  >
                    search
                  </i>
                </label>
                <i
                  class="clear-icon material-icons"
                >
                  close
                </i>
              </div>
            </form>
          </div>
        </nav>
      </div>
      <div
        class="filter-dropdown col s4 m2 l2"
      >
        <a
          class="dropdown-trigger btn-floating btn-large"
          data-target="dropdown1"
        >
          <i
            class="large material-icons"
          >
            sort
          </i>
        </a>
        <ul
          class="dropdown-content"
          id="dropdown1"
          tabindex="0"
        >
          <li
            class=""
            tabindex="0"
          >
            <a>
              <i
                class="material-icons"
              >
                access_time
              </i>
              By Created On
            </a>
          </li>
          <li
            class=""
            tabindex="0"
          >
            <a>
              <i
                class="material-icons"
              >
                thumb_up
              </i>
              By Upvotes
            </a>
          </li>
          <li
            class="divider"
            tabindex="-1"
          />
          <li
            tabindex="0"
          >
            <a>
              <i
                class="material-icons"
              >
                cached
              </i>
              Reset
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="hack-ideas--cards"
    >
      <div
        class="card-panel red lighten-4 center-align"
      >
        No ideas to show.
      </div>
    </div>
    <div
      class="fixed-action-btn register-idea-btn"
    >
      <a
        class="btn-floating btn-large"
      >
        <i
          class="large material-icons"
        >
          add
        </i>
      </a>
    </div>
  </div>
</div>
`;

exports[`Main Component renders login form if current user is not defined 1`] = `
<div>
  <div
    class="card-panel login-form-card z-depth-3"
  >
    <div
      class="row valign-wrapper login-form-wrapper"
    >
      <div
        class="col s12 m12 l6"
      >
        <h2>
          { Hack Ideas }
        </h2>
        <h6>
          Enter With Employee Id
        </h6>
      </div>
      <form
        class="col s12 m12 l6"
        novalidate=""
      >
        <div
          class="row valign-wrapper"
        >
          <div
            class="input-field col s12"
          >
            <input
              class="validate tooltipped"
              data-error="Employee ID is required"
              data-position="bottom"
              data-tooltip="Enter your 6 digit employee ID"
              id="empId"
              required=""
              type="text"
              value=""
            />
            <label
              for="empId"
            >
              Employee ID
            </label>
          </div>
        </div>
        <div
          class="row"
        >
          <button
            class="col s6 m12 l12 btn btn-large waves-effect"
            disabled=""
            name="btn_login"
            type="submit"
          >
            Login
            <i
              class="material-icons right"
            >
              send
            </i>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
